fuel-p2p/src/service.rs

```diff
@@ -412,8 +416,8 @@ mod tests {
                             // verifies that we've got at least a single peer address to send message to
                             if !peer_addresses.is_empty() && !message_sent  {
                                 message_sent = true;
-                                let default_tx = FuelGossipsubMessage::NewTx(Transaction::default());
-                                node_a.publish_message(selected_topic.clone(), default_tx).unwrap();
+                                let default_tx = GossipsubBroadcastRequest::NewTx(Arc::new(Transaction::default()));
```

Comment:
We should also have tests that ensure message routing and serialization work as expected across _all_ network event types we have. Currently, it seems like we only exercise the `NewTx` and `RequestBlock` code paths in our tests.

Reply:
added tests for all Gossipsub Messages, as for Request-Response, currently we only support RequestBlock anyway